zulipterminal/ui_tools/messages.py

```diff
@@ -796,14 +806,58 @@ def update_message_author_status(self) -> bool:
 
         return author_is_present
 
+    def transform_content_alert_words(content: str, alerted_list: List[str]) -> Any:
```

Comment:
If this is translated from a server implementation, it would be useful to add a comment to note that, and in the commit text too.

That could refer to the file as it is now, but also a commit ref, so that if code changes later then it is easier to track what has changed in the original code.

Having this function in a commit on it's own first, with detailed tests to cover various cases, would be good before we hook it into the processing in a subsequent commit.